sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts

```diff
@@ -92,9 +92,39 @@ export abstract class SFTPIntegrationBase<DownloadType> {
    */
   protected async downloadResponseFileLines(
     remoteFilePath: string,
-  ): Promise<string[]> {
+  ): Promise<string[]>;
+
+  /**
+   * Downloads the file specified on 'fileName' parameter from the
+   * SFAS integration folder on the SFTP.
+   * @param remoteFilePath full remote file path with file name.
+   * @param checkIfFileExist when set to true, check if file
+   * exist before downloading it.
+   * @returns parsed records from the file.
+   */
+  protected async downloadResponseFileLines(
```

Comment:
Using the overload signature just for the sake of the optional parameter can be avoided in this scenario IMO.

Reply:
The over load is actually used to return different types here.

The main intention of using overload is the return type is now changed to `<string[] | false>` from `<string[]>`

So most of the method consumers are using a forEach on the method return and we need to type cast everywhere if we are not using the overload. 

![image](https://github.com/bcgov/SIMS/assets/54600590/ed398b1c-db75-4366-94db-3f420df335dc)

Reply:
but the 2nd overload signature is the same as the implementation signature right? so, does it make sense? and 1st overload signature is covered in the 2nd overload signature.

Reply:
As per my understanding and what I saw(in one of the sync up with @andrewsignori-aot) if we use override to an implementation, the implementation becomes as good as private method, i.e it cannot be accessed directly from outside the class. Hence we need an override to the actual method with same arguments and return type.

Reply:
When overloading the first signature is not supposed to be exposed. If I am not wrong it does not even show in the VS Code IntelliSense.

![image](https://github.com/bcgov/SIMS/assets/61259237/d0bf19d8-6165-4b8b-b81d-333e3284bb9c)

![image](https://github.com/bcgov/SIMS/assets/61259237/a609c99b-bed1-41b9-ac0b-c29fccb13680)

I missed the part that the first and the second methods have different return types, so I am good.